hw1/02-palindrome.js

```diff
@@ -1,3 +1,46 @@
-const elem = document.querySelector('input');
+const elem = document.querySelector("input");
 
-elem.addEventListener('input', handleInput);
+elem.addEventListener("input", checkPalindrome);
+
+function checkPalindrome() {
+  const value = document.querySelector("input").value;
+
+  if (value != "") {
+    if (isNaN(value)) {
+      changeStatus("Input is not a number, please try again.", false);
+      return 1;
+    }
+
+    if (value < 0) {
+      changeStatus("Cannot use negative numbers, please try again.", false);
+      return 1;
+    }
+  } else {
+    changeStatus("", false);
+    return 1;
+  }
+
+  const valueString = value.toString();
+  const valueReversed = valueString.split("").reverse().join("");
+
+  if (valueString === valueReversed) {
+    changeStatus("This is a palindrome.", true);
+  } else {
+    changeStatus("Not a palindrome.", false);
+  }
+
+  return 0;
+}
+
+function changeStatus(text, correct) {
+  const result = document.getElementById("result");
```

Comment:
Similarly, I believe you should be able to move this declaration outside of the function to reduce the times `getElementById` is called.